### PR TITLE
major overhaul powstreams

### PR DESCRIPTION
**Pull request by jread-usgs**

### CODE REVIEW

**Comment by aappling-usgs** [comment]
the leaflet references are probably worth updating to package style, if possible.

* https://travis-ci.org/USGS-R/powstreams/builds/71461165#L646
```
Found the following significant warnings:
  Warning: replacing previous import by ‘leaflet::%>%’ when loading ‘powstreams’
See ‘/Users/travis/build/USGS-R/powstreams/powstreams.Rcheck/00install.out’ for details.
```

* results of ctrl-shift-f:leaflet include a lot of leaflet::fun syntax

File: tests/testthat/test-sites.R [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
-  #expect_is(list_sites(with_timeseries = c('wtr', 'doobs')), 'character')
-  
-  #expect_is(download_watershed(site = "nwis_01408500"), 'character')
-  #expect_is(load_timeseries(site = "nwis_01408500", variable = 'doobs'), 'data.frame')
-  #expect_is(load_timeseries(site = 'nwis_01408500', variable = c('doobs','wtr')), 'data.frame')
-  
-  expect_is(site_location("nwis_11126000"), 'data.frame')
-  expect_error(site_location("11126000"))
-  expect_is(site_location(c("nwis_11126000", "nwis_09258980")), 'data.frame')
+  expect_is(site_location("nwis_11126000"), "unitted_data.frame")
```
Comment (aappling-usgs) [inline tests/testthat/test-sites.R]: is site_location a better name than find_site_coords? it'd be nice to have a unified renaming scheme plan - maybe create a new issue to design & discuss the powstreams API

File: DESCRIPTION [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Package: powstreams
 Type: Package
 Title: powstreams
-Version: 0.6.0
+Version: 0.7.0
 Date: 2014-12-08
```
Comment (aappling-usgs) [inline DESCRIPTION]: maybe update the date too

File: R/explore_map.R [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#' explore powstreams site metadata on a map
+#' 
+#' open an interactive map for site data
+#' 
+#' @param browse use browser for map rendering
+#' @export
+#' @importFrom shiny runApp
+#' @import leaflet
+explore_map <- function(browse=TRUE){
```
Comment (aappling-usgs) [inline R/explore_map.R]: :+1:

File: tests/testthat/test-sites.R (hunk 1 above)
Comment (jread-usgs) [inline tests/testthat/test-sites.R]: I don't think it is a better name. I just didn't make any changes (that I can remember...) to the existing names.

File: R/powstreams.R [hunk 4]
```diff
@@ -34,8 +34,17 @@
...
+#'   }
+#'   
+#' @section interactive data exploration:
+#'   
+#'   \itemize{
+#'   
+#'   \item \code{\link{explore_map}} - explore site metadata with a 
+#'   map interface in a web browser
+#'   
+#'   \item \code{\link{explore_timeseries}} - explore timeseries data with a
```
Comment (aappling-usgs) [inline R/powstreams.R]: :+1:

File: R/shiny_helpers.R [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#' helper function for shiny launch
+#' @export
+#' @importFrom mda.streams get_meta
+#' @keywords internal
+get_types <- function(){
```
Comment (aappling-usgs) [inline R/shiny_helpers.R]: so in this case 'types' is not the list of meta_types but the list of data columns within any of the metadata tables? naming is a little confusing to me

**Comment by jread-usgs** [comment]
I can clean up the ::leaflet stuff (wasn't sure initially how to do that since it is running from inst), but I can't find the clash w/ %>%. Perhaps because both dygraphs and leaflet support it?

File: R/upload_presentation.R [hunk 6]
```diff
@@ -2,14 +2,14 @@
...
 upload_presentation <- function(file, ...){
   
   session_check_reauth(...)
-  session = current_session()
   
-  pres_folder <- mda.streams:::get_presentation_id(session)
-  id_out <- item_upload_create(parent_id = pres_folder, file, session)
+  pres_folder <- locate_folder('presentations')
+  id_out <- item_upload_create(parent_id = pres_folder, file)
   url_out <- .url_from_id(id_out)
```
Comment (aappling-usgs) [inline R/upload_presentation.R]: there's also now mda.streams:::format_item, which can do this in a slightly fancier way. optional; i guess if we used that then we'd want to make it an exported function in mda.streams, etc.

Comment (aappling-usgs) [inline R/upload_presentation.R]: or we could ultimately move upload_presentation and upload_publication into mda.streams.

File: inst/shiny/map/server.R [hunk 7]
```diff
@@ -1,26 +1,22 @@
-library(mda.streams)
-library(unitted)
-source('helpers.R')
-
-
+library(leaflet)
```
Comment (aappling-usgs) [inline inst/shiny/map/server.R]: oh, now i see why the leaflet dependencies work this way.

**Comment by aappling-usgs** [comment]
re: the leaflet %>% clash...maybe it has to do with this line in inst/shiny/map/server.R?
```
leaflet::leaflet() %>%
```
could be that it's getting interpreted oddly by the check algorithm

**Comment by jread-usgs** [comment]
I think the clash is because I actually have `library(leaflet)` in the shiny scripts

**Comment by aappling-usgs** [comment]
i'm done with my read-through. looks good! this is exciting to have a collaborator-facing product in place.

**Comment by jread-usgs** [comment]
cooool. Merging after fresh build. Fixed the warning

**Comment by aappling-usgs** [comment]
looks good to me. ready to merge?

**Pull request merged**